pkg/plugin/query.go

```diff
@@ -402,11 +405,30 @@ func deriveSchema(buf []byte) (*snellerSchema, error) {
 		}
 	}
 
+	// Restore column order
+	index := 0
+	err = status.ResultSet.UnpackStruct(func(field ion.Field) error {
```

Comment:
`result_set` may not be present if the query does a `SELECT * ...`, so you'd only want to do this when `!ResultSet.IsEmpty()`.